File: product/src/main/java/com/ecommerce/product/DTO/ProductResponse.java
package com.ecommerce.product.DTO;

import com.ecommerce.product.Enum.ProductCategory;
import com.ecommerce.product.Enum.ProductStatus;
import com.ecommerce.product.entity.ProductImages;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
public class ProductResponse {

    private Integer productId;

    private String sku;

    private String title;

    private String description;

    private Double price;

    private Integer stock;

    private ProductStatus status;

    private LocalDateTime createdAt;

    private List<ProductCategory> categories;

    private List<ProductImageDetails> images;
}

File: frontend/ecommerce/src/app/homepage/DTO/ProductResponse.ts
import {ProductStatus} from '../../product/Enum/ProductStatus';
import {ProductCategory} from '../../product/Enum/ProductCategory';

export interface ProductImage {
  imageId: number;
  objectName: string;
  url:string;
}

export interface ProductDTO {
  productId: number;
  userId: number;
  sku: string;
  title: string;
  description: string;
  price: number;
  stock: number;
  status: ProductStatus;
  categories: ProductCategory[];
  images: ProductImage[];
}
